### PR TITLE
Remove webhooks for etcd statefulsets.

### PR DESCRIPTION
**Pull request by georgekuruvillak**
**What this PR does / why we need it**:
Since [etcd-druid](https://github.com/gardener/etcd-druid) shall be managing etcd, this PR removes the webhooks for mutating etcd statefulset resources. It adds logic to add storageclass and storagecapacity for etcd resources depending on the role.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: controllers/provider-aws/charts/provider-aws/values.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 image:
   repository: eu.gcr.io/gardener-project/gardener/gardener-extension-hyper
   tag: latest
-  pullPolicy: IfNotPresent
+  pullPolicy: Always
```
Comment (rfranzke) [inline controllers/provider-aws/charts/provider-aws/values.yaml]: Revert this change.

File: go.mod [hunk 2]
```diff
@@ -3,51 +3,62 @@ module github.com/gardener/gardener-extensions
 go 1.13
 
 require (
-	cloud.google.com/go v0.43.0
+	cloud.google.com/go/storage v1.0.0
```
Comment (rfranzke) [inline go.mod]: Why was this updated?

File: go.mod [hunk 3]
```diff
@@ -59,7 +70,9 @@ replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v12.2.0+incompatible
 	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.4.2
 	github.com/census-instrumentation/opencensus-proto v0.1.0-0.20181214143942-ba49f56771b8 => github.com/census-instrumentation/opencensus-proto v0.0.3-0.20181214143942-ba49f56771b8
+	github.com/gardener/etcd-druid => github.com/georgekuruvillak/etcd-druid-1 v0.0.0-20200106104847-f1d9b72827af
```
Comment (rfranzke) [inline go.mod]: Can we get rid of this replace directive?

File: go.mod [hunk 4]
```diff
@@ -59,7 +70,9 @@ replace (
...
 	github.com/gardener/external-dns-management => github.com/gardener/external-dns-management v0.0.0-20190927090840-6659f5a46d13
+	github.com/gardener/gardener => github.com/georgekuruvillak/gardener v0.0.0-20200123100617-cb8b9668ce75
```
Comment (rfranzke) [inline go.mod]: Can we get rid of this replace directive?

File: controllers/provider-aws/charts/provider-aws/values.yaml (hunk 1 above)
Comment (georgekuruvillak) [reply controllers/provider-aws/charts/provider-aws/values.yaml]: Done

File: go.mod (hunk 2 above)
Comment (georgekuruvillak) [reply go.mod]: `make revendor` updated this file.

File: go.mod (hunk 4 above)
Comment (rfranzke) [reply go.mod]: Please revendor gardener based onhttps://github.com/gardener/gardener/pull/1869

File: go.mod (hunk 3 above)
Comment (rfranzke) [reply go.mod]: ping

**Comment by rfranzke** [comment]
We have moved the extension controllers into dedicated repositories.

Please open a new PR if you have changes in folder different than `controllers` in this repository, and another PR with only your changes in the respective provider specific repositories:
* https://github.com/gardener/gardener-extension-provider-alicloud
* https://github.com/gardener/gardener-extension-provider-aws
* https://github.com/gardener/gardener-extension-provider-azure
* https://github.com/gardener/gardener-extension-provider-gcp
* https://github.com/gardener/gardener-extension-provider-openstack
* https://github.com/gardener/gardener-extension-provider-vsphere

Thank you and sorry for the inconveniences.
/close
